### ISSUE TITLE
When i join my server my cleint throws an error that i dont have the same mods as the server.

### ISSUE THREAD

**Issue by Ethmangameon** [Noaaan/MythicMetals#180]
i found this in my console
## Mismatched states for entries of 'minecraft:block'
   - Found 'mythicmetals:adamantite_block', expected 'alloy_forgery:adamantite_block_forge_controller'
   this makes no sense to me since they are both there

**Comment by Noaaan** [comment]
Did you resolve this issue? This was recently reported in Discord, and I am yet to find out why this happens

**Comment by BurnyLlama** [comment]
If you use DEUF (https://www.curseforge.com/minecraft/mc-mods/deuf-refabricated) it might be the source of the problem. I believe it can deduplicate items differenty on the server and client, causing the mismatches.

**Comment by Noaaan** [comment]
I believe this issue happens on Quilt mainly, as it is some kind of load-order issue. An issue is open on Quilted FAPI regarding it

**Comment by BurnyLlama** [comment]
I see, but removing DEUF fixes it.

**Comment by encode42** [comment]
Should this issue stay open until the problem with Quilt is resolved?

It's been a few months and no progress on the issue has been made, which is quite unfortunate. Would a temporary workaround in MM be possible? I have no idea what the issue actually is, or how complex these systems are, but if Quilt doesn't seem interested in fixing it, what else do we have?

**Comment by Noaaan** [comment]
This issue is not really reproducible. If it didn't happen randomly I would be happy to implement something for it. The only real solution/workaround would be to disable/remove the Admantite Alloy Forge entirely when playing using Quilt. You could do this with a moddata (data-driven) override, or I could add a config option, but even then it doesn't actually fix the problem, only the symptom.

**Comment by encode42** [comment]
> The only real solution/workaround would be to disable/remove the Admantite Alloy Forge entirely

Though this is unfortunate, I would prefer this over not being able to use the mod entirely. I have been able to reproduce this issue in two different mod packs, so I'll try to get some more information for you!

**Comment by encode42** [comment]
This might have something to do with whether there's an even or odd amount of mods installed.

I still have to do some more testing, but I've found that load order issues arise with mods that use the `adventure` library only when there's an odd amount of mods installed in my mod pack. I can add or remove any mod, all that matters is whether that number is divisible by two.

I'll check if this fixes the registry desync, but if not, it appears I've found a different issue with Quilt.
